transaction/core/src/account_keys.rs

```diff
@@ -360,10 +363,28 @@ impl AccountKey {
             fog_authority_sig: Default::default(),
         };
 
-        // Compute fog_authority_sig as a signature over self.fog_authority_key_fingerprint
+        // NOTE: The fog_authority_sig is deterministic due to using the private key hash as the rng seed
         if !self.fog_report_url.is_empty() {
-            // FIXME: FOG-106 fog_authority_sig should be a Schnorrkel sig using subaddress_view_private
-            result.fog_authority_sig.extend(&[9u8, 9u8, 9u8, 9u8]);
+            // Construct the fog authority signature over the fingerprint using the view privkey
+            let view_private = self.subaddress_view_private(index);
+            let scalar: &Scalar = view_private.as_ref();
+
+            // Create rng seed from hash of private key in order to create nonce and sign
+            let seed: [u8; 32] = Sha256::digest(&scalar.to_bytes()).into();
```

Comment:
at this place, I would want to append `self.fog_authority_key_fingerprint` to the scalar bytes, so that the digest depends on both, because that ensures that this seed is not reused for two different messages that we are signing, even with the same private key.

and it would be my preference to use sha3 instead of sha256, because of djb's comments in this paper: http://ed25519.cr.yp.to/ed25519-20110926.pdf

> Pseudorandom generation of `r`
>
> ...
> Standard  PRF  hypotheses  imply  that  this  pseudorandom  session  key `r` is indistinguishable from a truly random string generated independently for each M, so there is no loss of security. Well-known length-extension properties prevent secret-prefix SHA-512 from being a PRF, but also do not threaten the security of Ed25519-SHA-512, since `r` is not visible to the attacker. All remaining SHA-3  candidates  are  explicitly  designed  to  be  PRFs,  and  we  will  not  hesitate  to recommend Ed25519-SHA-3 after SHA-3 is standardized. It would of course alsobe safe to generaterwith a cipher such as AES, combined with standard PRF-stretching mechanisms to support a long input; but we prefer to reuse H to save area in hardware implementations.

Reply:
Re: appending the message - that is done - will look at switching to sha3 now!

Reply:
sha3 done!